frontend/src/components/DatePicker/components/DatePickerInput.tsx

```diff
@@ -51,11 +57,12 @@ export const DatePickerInput = forwardRef<object, 'input'>((_props, ref) => {
           placeholder={placeholder}
           maskPlaceholder={placeholder}
           ref={mergedInputRef}
-          {...fcProps}
+          {...filteredFcProps}
           borderRightRadius={0}
           onBlur={handleInputBlur}
           onClick={handleInputClick}
           isReadOnly={fcProps.isReadOnly || !allowManualInput}
+          {...(highContrast && { variant: 'highContrast' })}
```

Comment:
Note that this `highContrast` is actually passed via the provider.

Reply:
fixed in https://github.com/opengovsg/FormSG/pull/8251/commits/fef5d89dbce7f2cfc768db8517a5ca4bde276a93